src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+    gl.bindTexture(gl.TEXTURE_2D, colorTexture);
+    const colorFormat = this._glColorFormat();
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      colorFormat.internalFormat,
+      this.width * this.density,
+      this.height * this.density,
+      0,
+      colorFormat.format,
+      colorFormat.type,
+      null
+    );
+
+    // Create the depth texture
+    const depthTexture = gl.createTexture();
+    if (!depthTexture) {
+      throw new Error('Unable to create depth texture');
+    }
+    const depthFormat = this._glDepthFormat();
+    gl.bindTexture(gl.TEXTURE_2D, depthTexture);
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      depthFormat.internalFormat,
+      this.width * this.density,
+      this.height * this.density,
+      0,
+      depthFormat.format,
+      depthFormat.type,
+      null
+    );
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, this.framebuffer);
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.COLOR_ATTACHMENT0,
+      gl.TEXTURE_2D,
+      colorTexture,
+      0
+    );
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.DEPTH_ATTACHMENT,
+      gl.TEXTURE_2D,
+      depthTexture,
+      0
+    );
+
+    // Create separate framebuffer for antialiasing
+    if (this.antialias) {
+      this.colorRenderbuffer = gl.createRenderbuffer();
+      gl.bindRenderbuffer(gl.RENDERBUFFER, this.colorRenderbuffer);
+      gl.renderbufferStorageMultisample(
+        gl.RENDERBUFFER,
+        Math.min(4, gl.getParameter(gl.MAX_SAMPLES)),
+        colorFormat.internalFormat,
+        this.width * this.density,
+        this.height * this.density
+      );
+
+      this.depthRenderbuffer = gl.createRenderbuffer();
+      gl.bindRenderbuffer(gl.RENDERBUFFER, this.depthRenderbuffer);
+      gl.renderbufferStorageMultisample(
+        gl.RENDERBUFFER,
+        Math.min(4, gl.getParameter(gl.MAX_SAMPLES)),
+        depthFormat.internalFormat,
+        this.width * this.density,
+        this.height * this.density
+      );
+
+      gl.bindFramebuffer(gl.FRAMEBUFFER, this.aaFramebuffer);
+      gl.framebufferRenderbuffer(
+        gl.FRAMEBUFFER,
+        gl.COLOR_ATTACHMENT0,
+        gl.RENDERBUFFER,
+        this.colorRenderbuffer
+      );
+      gl.framebufferRenderbuffer(
+        gl.FRAMEBUFFER,
+        gl.DEPTH_ATTACHMENT,
+        gl.RENDERBUFFER,
+        this.depthRenderbuffer
+      );
+    }
+
+    this.depthTexture = depthTexture;
+    this.colorTexture = colorTexture;
+
+    this.depth = new FramebufferTexture(this, 'depthTexture');
+    const depthP5Texture = new p5.Texture(
+      this.target._renderer,
+      this.depth,
+      {
+        minFilter: gl.NEAREST,
+        magFilter: gl.NEAREST
+      }
+    );
+    this.target._renderer.textures.set(this.depth, depthP5Texture);
+
+    this.color = new FramebufferTexture(this, 'colorTexture');
+    const colorP5Texture = new p5.Texture(
+      this.target._renderer,
+      this.color,
+      {
+        glMinFilter: gl.LINEAR,
```

Comment:
Same Q as above. I think there is a strong argument to be made for having control over the interpolation mode here. Framebuffers are often used to store data, using linear scaling would corrupt that data.

Reply:
True, do you think we should add these as options you can supply in settings?

Reply:
If we didn't supply them as settings, would the method for changing the interpolation mode to be to call those methods yourself on the textures after the framebuffer has been constructed?

Personally I think they'd be useful settings in the constructor but I know there are a lot of possible values there already. If we added settings for interpolation, then we might also want to add settings for texture wrap (REPEAT, CLAMP, MIRROR). If you give a mouse a cookie...

Reply:
I think I'm in favour of adding it to the constructor -- it's probably not the end of the world having those all in the constructor since they're all optional and we're using a settings object, so you can just specify the bits you want. I think probably the existing `textureWrap()` function will let people set the wrap mode so we can keep that out for now, although I think adding it to the constructor in the future could make sense too.

Reply:
I'm also thinking, for simplicity should we assume that the min and mag filter should be the same for now?